src/eventbus.js

```diff
@@ -29,16 +29,8 @@ class EventBus {
             return; // If not just return
         }
 
-        let newArray = []; // Array that will contain all events except the one removed
-        for (const event of this.events[type]) { // Iterate all events of the specified type
-            // Check if the event of the current iteration and the passed one are equal
-            const isSame = event.scope == scope && event.callback == callback;
-            if (!isSame) { // If they are not the same...
-                newArray.push(event); // ...push the current event to the newArray
-            }
-        }
-
-        this.events[type] = newArray; // Reset the events array with all events except the removed one
+        // keep all elements that aren't equal to the passed event
+        this.events[type] = this.events[type].filter(event => event.scope !== scope || event.callback !== callback);
```

Comment:
Despite the fact that the code is much shorter I'm not sure if it's much clearer too.

Reply:
We could split up the filtering logic and the execution itself like this:  
```suggestion
        const filterFn = event => event.scope !== scope || event.callback !== callback;
        this.events[type] = this.events[type].filter(filterFn);
```

I personally like the array methods like `.filter()` because they explain very clearly whats going on (getting a new filtered array) and we dont have to implement how its done, just what the filter-criteria are.

The same goes for `.some()`. We dont have to implement the logic how its done, just what we are looking for in the array.

Reply:
Splitting up the filtering and execution logic sounds good.

Can you make a PR with changes as proposed by you?